refocus-bdk-server.js

```diff
@@ -887,6 +888,30 @@ module.exports = (config) => {
         bot + '&name=' + name, TOKEN, DEFAULT_TRIES, log, PROXY_URL);
     }, // getBotData
 
+    /**
+     * Generic function for getting and parsing bot data in JSON form.
+     *
+     * @param {String} roomId - the id of the room to query
+     * @param {String} botDataName - the name of the bot data to query
+     * @param {String} bot - the name of the bot
+     * @returns {Object} - the parsed bot data
+     */
+    getAndParseBotData(roomId, botDataName, bot) {
+      if (!roomId || !botDataName) return null;
+      let retBotData = null;
+      try {
+        return this.getBotData(roomId, bot, botDataName).then((botData) => {
+          if (botData && botData.body && botData.body.length > ZERO) {
+            retBotData = JSON.parse(botData.body[ZERO].value);
+          }
+          return retBotData;
+        });
+      } catch (err) {
+        log.error(err);
+        return null;
```

Comment:
If we get a null value, we probably don't know wether the botData is missing (need to be created) or an error occur during parsing. Will it be a concern for the current usage in bots?

Reply:
for current usage no but in future could probably make function better by creating a bot data if none exists